Data/Text/Lazy.hs

```diff
@@ -929,6 +939,21 @@ replicate n t
                  | otherwise = t : rep (i+1)
 {-# INLINE [1] replicate #-}
 
+-- | 'cycle' ties a finite 'Text' into a circular one, or equivalently,
```

Comment:
I'd suggest to add the non-empty precondition to the documentation as follows.

'cycle' ties a finite, non-empty 'Text' into a circular one ....